onedal/datatypes/data_conversion_sua_iface.cpp

```diff
@@ -0,0 +1,157 @@
+/*******************************************************************************
+* Copyright 2023 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#ifdef ONEDAL_DATA_PARALLEL
+#define NO_IMPORT_ARRAY
+
+#include <stdexcept>
+#include <string>
+#include <utility>
+
+#include "oneapi/dal/common.hpp"
+#include "oneapi/dal/detail/common.hpp"
+#include "oneapi/dal/table/homogen.hpp"
+#include "oneapi/dal/table/detail/homogen_utils.hpp"
+
+#include "onedal/common/policy_common.hpp"
+#include "onedal/datatypes/data_conversion_sua_iface.hpp"
+#include "onedal/datatypes/utils/dtype_conversions.hpp"
+#include "onedal/datatypes/utils/dtype_dispatcher.hpp"
+#include "onedal/datatypes/utils/sua_iface_helpers.hpp"
+
+namespace oneapi::dal::python {
+
+template <typename Type>
+dal::table convert_to_homogen_impl(py::object obj) {
+    auto sua_iface_dict = get_sua_interface(obj);
+
+    const auto deleter = [obj](const Type*) {
```

Comment:
For example I know what this does because of its importance towards the python object memory model, but may not be directly clear to someone looking at this with fresh eyes.  Could you go through the cpp code and add a lot of comments/ function documentation? It  would be helpful in reviewing and future reference.

Reply:
Comments are added.